ngraph/frontend/ir/src/frontend.cpp

```diff
@@ -2,70 +2,169 @@
-            exts = ov::as_type_ptr<ov::VariantWrapper<std::vector<InferenceEngine::IExtensionPtr>>>(variant)->get();
+        if (ov::is_type<ov::VariantWrapper<ov::ModelPath>>(variant)) {
+            model_path = ov::as_type_ptr<ov::VariantWrapper<ov::ModelPath>>(variant)->get().get_model_path();
+            local_model_stream.open(model_path, std::ios::in | std::ifstream::binary);
+        } else if (ov::is_type<ov::VariantWrapper<std::istream*>>(variant)) {
+            provided_model_stream = ov::as_type_ptr<ov::VariantWrapper<std::istream*>>(variant)->get();
+        } else if (ov::is_type<ov::VariantWrapper<std::istringstream*>>(variant)) {
+            provided_model_stream = ov::as_type_ptr<ov::VariantWrapper<std::istringstream*>>(variant)->get();
+        } else if (ov::is_type<ov::VariantWrapper<ov::WeightsPath>>(variant)) {
+            weights_path = ov::as_type_ptr<ov::VariantWrapper<ov::WeightsPath>>(variant)->get().get_weights_path();
+        } else if (ov::is_type<ov::ConstantVariant>(variant)) {
+            weights = ov::as_type_ptr<ov::ConstantVariant>(variant)->get();
+        } else if (ov::is_type<ov::ExtensionsVariant>(variant)) {
+            extensions = ov::as_type_ptr<ov::ExtensionsVariant>(variant)->get();
+        }
+    }
+
+    if (weights_path.empty()) {
+        auto pos = model_path.rfind('.');
+        if (pos != std::string::npos)
+            weights_path = model_path.substr(0, pos);
+
+        weights_path += ".bin";
+        if (!FileUtils::fileExist(weights_path)) {
+            weights_path.clear();
         }
     }
-    return std::make_shared<InputModelIR>(root, weights, exts);
+
+    if (!weights_path.empty()) {
+        std::ifstream bin_stream;
+        bin_stream.open(weights_path, std::ios::binary);
+        if (!bin_stream.is_open())
+            IR_THROW("Weights file " + weights_path + " cannot be opened!");
+
+        bin_stream.seekg(0, std::ios::end);
+        size_t file_size = bin_stream.tellg();
+        bin_stream.seekg(0, std::ios::beg);
+
+        weights = std::make_shared<ngraph::op::Constant>(ngraph::element::u8, Shape{file_size}, 0);
+
+        // TODO: get rid of const_cast
+        bin_stream.read(const_cast<char*>(weights->get_data_ptr<char>()), file_size);
+        bin_stream.close();
+    }
+
+    if (provided_model_stream != nullptr) {
+        return std::make_shared<InputModelIR>(*provided_model_stream, weights, extensions);
+    } else if (local_model_stream.is_open()) {
+        const auto input_model = std::make_shared<InputModelIR>(local_model_stream, weights, extensions);
+        local_model_stream.close();
+        return input_model;
+    }
+    return nullptr;
 }
 
-std::shared_ptr<ngraph::Function> FrontEndIR::convert(InputModel::Ptr model) const {
+void FrontEndIR::normalize(std::shared_ptr<ngraph::Function> function) const {
+    ngraph::pass::Manager m;
+    m.register_pass<ngraph::pass::MOCTransformations>(false);
+    m.run_passes(function);
+}
```

Comment:
why IR is not normalized after MO?

Reply:
I doublechecked and convert method which is supposed to return normalize Function applies normalization.

Reply:
yes, but we actually need any normalization after MO? I understand that the same FE is used in MO and IE, but pipelines can / should be different:
* MO:
  * converts from original FW using C++ FW or native FW 
  * applies normalization using either C++ FE or MOC passes in case of python FE
  * writes normalized IR to file
* IR FE:
  * just reads IR
  * why do we need normalization? MO applies it already on the second step. The only reason I see - if we normalize IR produced by older releases, is it the case?

Reply:
Yes, IR produce by older version of MO needs to be normalized. But the current final* solution is a bit different so MOC transformations will be applied in plugin pipelines but not in FE normalization step due to some inconsistency and potential double transformations execution in runtime. 
_* - if we find better solution we will change it but now this solution suits to the all use-cases_